packages/next/src/utils.ts

```diff
@@ -2858,7 +2858,12 @@ export const onPrerenderRoute =
         (r): r is RoutesManifestRoute =>
           r.page === pageKey && !('isMiddleware' in r)
       ) as RoutesManifestRoute | undefined;
-      const isDynamic = isDynamicRoute(routeKey);
+      const isRouteKeyDynamic =
+        routesManifest?.dynamicRoutes.some(
+          (r): r is RoutesManifestRoute =>
+            r.page === routeKey && !('isMiddleware' in r)
+        ) ?? false;
+      const isDynamic = isRouteKeyDynamic || isDynamicRoute(routeKey);
```

Comment:
does this handling need to be baked into `isDynamicRoute`? I'm wondering if there are other cases that are are incorrectly marking these routes as static when they should be dynamic. 

I suspect that util came from Next.js and has drifted as implementations changed (eg the one in Next has handling for interception routes: https://github.com/vercel/next.js/blob/13174c55256308ba6864ce7cd3fe53e30f2cd7dd/packages/next/src/shared/lib/router/utils/is-dynamic.ts)

I feel like we need to either fully rely on a util, or fully rely on Next.js telling us something's dynamic. Feels strange to do both